src/test/java/org/folio/rest/impl/EncumbrancesTest.java

```diff
@@ -0,0 +1,646 @@
+    assertEquals(fromBudgetBefore.getAvailable() , fromBudgetAfterUpdate.getAvailable());
+    assertEquals(fromBudgetBefore.getUnavailable(), fromBudgetAfterUpdate.getUnavailable());
+    assertEquals(fromBudgetBefore.getAwaitingPayment(), fromBudgetAfterUpdate.getAwaitingPayment());
+
+  }
+
+  @Test
+  void testUpdateEncumbranceNotFound() throws MalformedURLException {
+
+    String orderId = UUID.randomUUID().toString();
+    createOrderSummary(orderId, 2);
+
+    JsonObject jsonTx = new JsonObject(getFile(ENCUMBRANCE_SAMPLE));
+    jsonTx.remove("id");
+    Transaction encumbrance = jsonTx.mapTo(Transaction.class);
+    encumbrance.getEncumbrance().setSourcePurchaseOrderId(orderId);
+
+    // Try to update non-existent transaction
+    putData(TRANSACTION.getEndpointWithId(), UUID.randomUUID().toString(), JsonObject.mapFrom(encumbrance).encodePrettily(), TRANSACTION_TENANT_HEADER).then().statusCode(404);
+
+  }
+
+  @Test
+  void tesPostEncumbranceUnavailableMustNotIncludeOverEncumberedAmounts() throws MalformedURLException {
+
+    String fiscalYearId = createFiscalYear();
+    String ledgerId = createLedger(fiscalYearId, true);
+    String fundId = createFund(ledgerId);
+
+    Budget budgetBefore = buildBudget(fiscalYearId, fundId);
+
+    String budgetId = postData(BUDGET.getEndpoint(), JsonObject.mapFrom(budgetBefore).encodePrettily(), TRANSACTION_TENANT_HEADER).then()
+      .statusCode(201).extract().as(Budget.class).getId();
+
+    String ledgerFYEndpointWithQueryParams = String.format(LEDGER_FYS_ENDPOINT, ledgerId, fiscalYearId);
+
+    LedgerFY ledgerFYBefore = getLedgerFYAndValidate(ledgerFYEndpointWithQueryParams);
+
+    String orderId = UUID.randomUUID().toString();
+    createOrderSummary(orderId, 1);
+
+    JsonObject jsonTx = new JsonObject(getFile(ENCUMBRANCE_SAMPLE));
+    jsonTx.remove("id");
+    Transaction encumbrance = jsonTx.mapTo(Transaction.class);
+
+    encumbrance.getEncumbrance().setSourcePurchaseOrderId(orderId);
+    encumbrance.setFromFundId(fundId);
+    encumbrance.setFiscalYearId(fiscalYearId);
+    encumbrance.setSourceFiscalYearId(null);
+
+    final double transactionAmount = 10000d;
+    encumbrance.setAmount(transactionAmount);
+
+    String transactionSample = JsonObject.mapFrom(encumbrance).encodePrettily();
+
+    String encumbranceId = postData(TRANSACTION.getEndpoint(), transactionSample, TRANSACTION_TENANT_HEADER).then()
+      .statusCode(201).extract().as(Transaction.class).getId();
+
+    // encumbrance do not appear in transaction table
+    getDataById(TRANSACTION.getEndpointWithId(), encumbranceId, TRANSACTION_TENANT_HEADER).then().statusCode(200);
+
+    Budget budgetAfter = getDataById(BUDGET.getEndpointWithId(), budgetId, TRANSACTION_TENANT_HEADER).then()
+      .statusCode(200).extract().as(Budget.class);
+
+    Double expectedOverEncumbrance = subtractValues(transactionAmount, budgetBefore.getAvailable());
+    assertEquals(budgetBefore.getAllocated(), budgetAfter.getUnavailable());
```

Comment:
While these assertions aren't incorrect, it wouldn't hurt to add some that explicitly test the equations... e.g.

```
assertEquals(budgetAfter.getOverEncumbrance(), Math.max(0, budgetAfter.getEncumbered()-Math.max(0, budgetAfter.getAllocated()-budgetAfter.getAwaitingPayment()-budgetAfter.getExpenditures())));
...
assertTrue(budgetAfter.getUnavailable() >= 0);
assertEquals(budgetAfter.getUnavailable(), budgetAfter.getEncumbered() + budgetAfter.getAwaitingPayment() + budgetAfter.getExpenditures() - budgetAfter.getOverEncumbrance() - budgetAfter.getOverExpended());
```

In fact adding a method that takes a budget and validates it against these equations might be a good idea, as it would be an easy way to add sanity checks to additional test cases.